webdriver/tests/bidi/browsing_context/navigate/navigate.py

```diff
@@ -108,3 +109,44 @@ async def test_same_document_navigation_in_before_unload(bidi_session, new_tab,
 
     url_after = url_before.replace("empty.html", "other.html")
     await navigate_and_assert(bidi_session, new_tab, url_after, "complete")
+
+
+@pytest.mark.parametrize(
+    "script",
+    [
+        "<script>window.location='{url}'</script>",
+        """<script>window.addEventListener('DOMContentLoaded', () => {{
+            window.location = '{url}';
+        }});</script>""",
+        """<script>window.addEventListener('load', () => {{
+            window.location = '{url}';
+       }});</script>""",
+    ],
+    ids=["Redirect immediately", "Redirect on DOMContentLoaded", "Redirect on load"],
+)
+@pytest.mark.parametrize("wait", ["none", "interactive", "complete"])
+async def test_interrupted_navigation(
+    bidi_session,
+    subscribe_events,
+    new_tab,
+    url,
+    inline,
+    wait_for_event,
+    wait_for_future_safe,
+    script,
+    wait,
+):
+    url_after = url("/webdriver/tests/bidi/browsing_context/support/empty.html")
+    url_before = inline(script.format(url=url_after))
+
+    await subscribe_events(events=[CONTEXT_LOAD_EVENT])
+    on_entry = wait_for_event(CONTEXT_LOAD_EVENT)
+
+    result = await bidi_session.browsing_context.navigate(
```

Comment:
unrelated to this specific test: do we actually have a way to trigger an error with navigate? I guess a navigate call with beforeunload handler navigating elsewhere would trigger the condition of a new navigation before a commit?

Reply:
That should work. In practice any scenario that notifies BiDi about a "canceled"/"navigation failed" status. From a quick read of the HTML spec I can see: CSP issues, beforeunload, unloading a tab, closing/destroying a document.

Eg wait=complete + closing a tab before the navigation finishes should lead to an error.